core/src/main/java/de/bwaldvogel/mongo/oplog/OplogDocument.java

```diff
@@ -0,0 +1,130 @@
+package de.bwaldvogel.mongo.oplog;
+
+import java.time.Instant;
+import java.util.UUID;
+
+import de.bwaldvogel.mongo.bson.Bson;
+import de.bwaldvogel.mongo.bson.BsonTimestamp;
+import de.bwaldvogel.mongo.bson.Document;
+
+public class OplogDocument implements Bson {
+    private final Document document;
+
+    public OplogDocument() {
+        document = new Document();
+        withProtocolVersion(2L);
+    }
+
+    public Document asDocument() {
+        return document;
+    }
+
+    public BsonTimestamp getTimestamp() {
+        return (BsonTimestamp) document.get("ts");
+    }
+
+    public OplogDocument withTimestamp(BsonTimestamp timestamp) {
+        document.put("ts", timestamp);
+        return this;
+    }
+
+    public long getT() {
+        return (Long) document.get("t");
+    }
+
+    public OplogDocument withT(long t) {
+        document.put("t", t);
+        return this;
+    }
+
+    public long getHash() {
+        return (Long) document.get("h");
+    }
+
+    public OplogDocument withHash(long h) {
+        document.put("h", h);
+        return this;
+    }
+
+    public long getProtocolVersion() {
+        return (Long) document.get("v");
+    }
+
+    public OplogDocument withProtocolVersion(long protocolVersion) {
+        document.put("v", protocolVersion);
+        return this;
+    }
+
+    public OperationType getOperationType() {
+        return OperationType.valueOf((String) document.get("op"));
```

Comment:
I think it should be `OperationType.fromCode()`

Reply:
I think so as well. Fixed in e08e91cdc6129c85e3f07b2a0ef6f474e5628a6e